aiplatform/snippets/create_dataset_test.go

```diff
@@ -0,0 +1,105 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Prompt: Write a test for createDataset in createDataset.go
+
+package snippets
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"log"
+	"strings"
+	"testing"
+
+	aiplatform "cloud.google.com/go/aiplatform/apiv1"
+	aiplatformpb "cloud.google.com/go/aiplatform/apiv1/aiplatformpb"
+	"github.com/GoogleCloudPlatform/golang-samples/internal/testutil"
+	"google.golang.org/api/option"
+)
+
+var (
+	datasetID string
+	output    string
+	projectID string
+	region    string = "us-central1"
+)
+
+func TestMain(m *testing.M) {
+	tc, ok := testutil.ContextMain(m)
+
+	projectID = tc.ProjectID
+
+	if !ok {
+		log.Fatal("couldn't initialize test")
+		return
+	}
+
+	m.Run()
+
+	deleteDataset()
+}
+
+func TestCreateDataset(t *testing.T) {
+	tc := testutil.SystemTest(t)
+	datasetID = "my-image-dataset"
```

Comment:
Can we add a dynamic component to this dataset, like some UUID or something? It will help reduce resource contention.

Reply:
That is a reasonable request ... unfortunately, the developer doesn't control the actual dataset ID. It is automatically assigned by the service. I will rename this variable to reflect that behavior.